src/main/java/org/springframework/samples/petclinic/rest/importcsv/ImportCSV.java

```diff
@@ -26,114 +26,198 @@
     @Autowired
     private ClinicService clinicService;
 
+    
+    /** importPets: Parses a csv message and adds/deletes pets from the clinicService
+     * @param csv
+     * @return ResponseEntity<List<Pet>>
+     */
     @PreAuthorize("hasRole(@roles.OWNER_ADMIN)")
     @RequestMapping(value = "importPets",
         method = RequestMethod.POST,
         consumes = "text/plain",
         produces = "application/json")
     public ResponseEntity<List<Pet>> importPets(@RequestBody String csv) {
-
-        int i = 0;
-        List<Pet> pets = new LinkedList<Pet>();
-        Pet pet;
-
-        do {
-            pet = new Pet();
-
-            String field = "";
-            while (i < csv.length() && csv.charAt(i) != ';') {
-                field += csv.charAt(i++);
+        // the method has to renamed - it doesn't just import, it also deletes
+        List<Pet> petsToAdd = new LinkedList<>();
+        List<Pet> petsToDelete = new LinkedList<>();
+        for (String line : csv.split("\\r?\\n")) {
```

Comment:
you could think about extracting the most contents of `importPets` to a new method, so `importPets` is only responsible about handling the request and the new method takes care of orchestrating the parsing and so on (probably you would have done this with more time :) )